lib/github-pages-health-check/errors/invalid_aaaa_record_error.rb

```diff
@@ -0,0 +1,19 @@
+# frozen_string_literal: true
+
+module GitHubPages
+  module HealthCheck
+    module Errors
+      class InvalidAAAARecordError < GitHubPages::HealthCheck::Error
+        DOCUMENTATION_PATH = "/articles/setting-up-a-custom-domain-with-github-pages/".freeze
+
+        def message
+          <<-MSG
+             Your site's DNS settings are using a custom subdomain, #{domain.host},
+             that's set up with an AAAA record. GitHub pages currently does not support
+             ipv6 and this AAAA record will interfere with https certificate creation.
```

Comment:
`IPv6`, and `HTTPS`

Not sure if we should say anything about HTTPS directly (though it applies!) because having IPv6 records could actually cause issues serving for clients that lookup IPv6 first before IPv4. So it could be a bigger issue.